preflight just fails on container image test

## Bug Description
preflight fails with error on RHEL 8.6

## Version and Command Invocation
preflight version 1.3.2 <commit: cfedcc7a46d47f9679eb137751ea22750d623c72>

## Steps to Reproduce:
(How can we reproduce this?)
1) Run it against any container image

## Expected Result
expect to pass, which is does but then it throws errors at the end of the test


## Actual Result
```
$ preflight-linux-amd64 check container <myimage>
time="2022-07-20T13:25:57-04:00" level=info msg="certification library version 1.3.2 <commit: cfedcc7a46d47f9679eb137751ea22750d623c72>"
time="2022-07-20T13:26:32-04:00" level=info msg="check completed: HasLicense" result=PASSED
time="2022-07-20T13:26:32-04:00" level=info msg="check completed: HasUniqueTag" result=PASSED
time="2022-07-20T13:26:32-04:00" level=info msg="check completed: LayerCountAcceptable" result=PASSED
time="2022-07-20T13:26:32-04:00" level=info msg="check completed: HasNoProhibitedPackages" result=PASSED
time="2022-07-20T13:26:32-04:00" level=info msg="check completed: HasRequiredLabel" result=PASSED
time="2022-07-20T13:26:32-04:00" level=info msg="USER steamuser specified that is non-root"
time="2022-07-20T13:26:32-04:00" level=info msg="check completed: RunAsNonRoot" result=PASSED
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/policy/container.(*HasModifiedFilesCheck).dropFirstLayerIfEmpty(...)
	/go/src/preflight/certification/internal/policy/container/has_modified_files.go:108
github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/policy/container.(*HasModifiedFilesCheck).getDataToValidate(0x471753, {0x1b36f98, 0xc00011c000}, {{0x7ffcdd3cd4a4, 0x49}, {0xc0006f4400, 0x1c}, {0x1b6c248, 0xc0003ca180}, {0x7ffcdd3cd4b8, ...}, ...})
	/go/src/preflight/certification/internal/policy/container/has_modified_files.go:87 +0x557
github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/policy/container.(*HasModifiedFilesCheck).Validate(0x60, {0x1b36f98, 0xc00011c000}, {{0x7ffcdd3cd4a4, 0x49}, {0xc0006f4400, 0x1c}, {0x1b6c248, 0xc0003ca180}, {0x7ffcdd3cd4b8, ...}, ...})
	/go/src/preflight/certification/internal/policy/container/has_modified_files.go:31 +0x5e
github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/engine.(*CraneEngine).ExecuteChecks(0xc000136a00, {0x1b36f98, 0xc00011c000})
	/go/src/preflight/certification/internal/engine/engine.go:180 +0x162e
github.com/redhat-openshift-ecosystem/openshift-preflight/cmd.preflightCheck({0x1b36f98, 0xc00011c000}, 0xc000028360, {0x0, 0x0}, {0x1b1fdf8, 0xc000136a00}, {0x1b24f30, 0xc0001c0de0}, {0x1b24f90, ...}, ...)
	/go/src/preflight/cmd/preflight_check.go:43 +0x2c2
github.com/redhat-openshift-ecosystem/openshift-preflight/cmd.checkContainerRunE(0x2774700, {0xc0004a76c0, 0x1, 0x1})
	/go/src/preflight/cmd/check_container.go:99 +0x32a
github.com/spf13/cobra.(*Command).execute(0x2774700, {0xc0004a76a0, 0x1, 0x1})
	/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x2774e80)
	/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:895
github.com/redhat-openshift-ecosystem/openshift-preflight/cmd.Execute()
	/go/src/preflight/cmd/root.go:38 +0x58
main.main()
	/go/src/preflight/main.go:8 +0x17
```


## Additional Context
Running this on RHEL 8.6

--- a/certification/internal/policy/container/has_modified_files.go
+++ b/certification/internal/policy/container/has_modified_files.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"io"
 	"strings"
+	"sync"
 
 	rpmdb "github.com/knqyf263/go-rpmdb/pkg"
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
@@ -67,6 +68,8 @@ func (p *HasModifiedFilesCheck) getDataToValidate(ctx context.Context, imgRef ce
 		}
 		pathChan := make(chan string)
 
+		var wg sync.WaitGroup
+		wg.Add(1)
 		go func() {
 			// For each path in the pathChan, add it to the layer's
 			// list of files.
@@ -76,12 +79,15 @@ func (p *HasModifiedFilesCheck) getDataToValidate(ctx context.Context, imgRef ce
 			}
 			// also add it to the overall list of files.
 			files = append(files, layerFiles)
+			wg.Done()
 		}()
 		// add paths to the pathChan
 		err = untar(pathChan, r)
 		if err != nil {
 			return nil, fmt.Errorf("failed to extract tarball: %w", err)
 		}
+
+		wg.Wait() // wait for file list to get appended to the files var
 	}
 
 	files, dropped := p.dropFirstLayerIfEmpty(files)

